src/main/java/net/minestom/server/entity/Entity.java

```diff
@@ -532,9 +609,17 @@ public void tick(long time) {
                 Vector newVelocityOut = new Vector();
 
                 // Gravity force
-                final double gravityY = !hasNoGravity() ? Math.min(
-                        gravityDragPerTick + (gravityAcceleration * (double) gravityTickCount),
-                        gravityTerminalVelocity) : 0;
+                double gravityY;
```

Comment:
I'd suggest creating a 'protected getGravityValue()' with this code, so it is properly separated from the rest and can be overridden to allow custom gravity algorithms

Reply:
That's a good idea, moved it to a method (and left only the has gravity check there).